modules/sor/sorV2/sorV2.service.ts

```diff
@@ -0,0 +1,243 @@
+                    tIn,
+                    tOut,
+                    this.mapSwapType(swapType),
+                    swapAmount,
+                    poolsFromDb,
+                    // swapOptions, // TODO - Handle properly
+                );
+
+        if (!swap) throw new Error('Swap is undefined');
+        console.log(`Swap (db pools)`);
+        console.log(poolsFromDb.length);
+        console.log(swap.swaps);
+        console.log(swap.outputAmount.amount.toString());
+        console.log(`------------------`);
+
+        // Just using to compare results against onchain pool calls
+        // await getSwapCompare(tIn, tOut, swapAmount);
+
+        // TODO - Update with proper required result data
+        return {
+            tokenIn,
+            tokenOut,
+            result: swap.outputAmount.amount.toString()
+        }  
+    }
+
+    public mapResultToCowSwap(swap: string): string {
+        // TODO - match existing CowSwap SOR API format so its plug and play
+        return swap;
+    }
+
+    /**
+     * Gets a b-sdk Token based off tokenAddr.
+     * @param tokenAddr 
+     * @param chainId 
+     * @returns 
+     */
+    private async getToken(tokenAddr: Address, chainId: ChainId): Promise<Token> {
+        const tokens = await tokenService.getTokens();
+        const prismaToken = this.getPrismaToken(tokenAddr, tokens);
+        return new Token(
+            chainId,
+            tokenAddr,
+            prismaToken.decimals,
+            prismaToken.symbol,
+        );
+    }
+
+    private getPrismaToken(tokenAddress: string, tokens: PrismaToken[]): PrismaToken {
+        tokenAddress = tokenAddress.toLowerCase();
+        const match = tokens.find((token) => token.address === tokenAddress);
+
+        if (!match) {
+            throw new Error('Unknown token: ' + tokenAddress);
+        }
+        return match;
+    }
+
+    private mapSwapType(swapType: GqlSorSwapType): SwapKind {
+        return swapType === "EXACT_IN" ? SwapKind.GivenIn : SwapKind.GivenOut;
+    }
+
+    /**
+     * Fetch pools from Prisma and map to b-sdk BasePool.
+     * @returns 
+     */
+    private async getBasePoolsFromDb(): Promise<BasePool[]> {
+        const pools = await prisma.prismaPool.findMany({
+            where: { 
+                chain: networkContext.chain,
+                dynamicData: {
+                    totalSharesNum: {
+                        gt: 0.000000000001,
+                    },
+                    swapEnabled: true,
+                },
+            },
+            include: prismaPoolWithDynamic.include
+        });
+        const rawPools = this.mapToRawPools(pools);
+        // TODO - The pools below cause issue with b-sdk maths. Both have a token with balance = 0 so maybe that's issue?
+        const linearPoolsToIgnore = [
+            "0xbfa413a2ff0f20456d57b643746133f54bfe0cd20000000000000000000004c3", 
+            "0xdc063deafce952160ec112fa382ac206305657e60000000000000000000004c4", 
+        ]
```

Comment:
These would be better configured in the network config. We also have some pools on FTM that we want to exclude from the SOR routing. Also it can be built into the prisma query.

Reply:
I'm hoping we can fix whatever is causing these specific pools to throw so we don't need to exclude them but for now I've moved them as you suggested.

Reply:
My bet is that these are uninitialized linears that fail on the getRate call. But the QueriesHelperContract should actually handle the failed call at least.